controllers/userController.js

```diff
@@ -16,6 +17,51 @@ exports.getAllUsers = async (req, res) => {
   }
 }
 
+exports.getUsers = async (req, res) => {
+
+  //get the current user id from the token
+  const token = getEvenToken(req)
+  const decoded = decodeTkn(token, req.config.JWT_KEY)
+  const id = decoded[constants.idClaim]
+
+  //check if the current user is an admin
+  let isAdmin = req.user.admin
+  
+  if(!isAdmin) {
+    //get the data of the current user
+    try {
+      const userData = await req.db.User.findById(id)
+
+      return res.status(HttpStatus.OK).json({
+        success: true,
+        userData
+      })
+    } catch (error) {
+      req.log.error(`Unable to get current user -> ${req.url} -> ${error}`)
+      return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+        success: false
+      })
+    }
+  }
+
+  if(isAdmin) {
+    //get the data of all users based on query parameters
+    try {
+      const users = await req.db.User.find(req.query)
```

Comment:
please make sure this is working even tho I pass no data in query params.

Reply:
Made sure it has the expected behavior. It returns all the users